### DISCUSSION TITLE
modbus comms changes from 2.4.6 to 2.6.0

### DISCUSSION THREAD

**Discussion by BruciusL** [artisan-roaster-scope/artisan#846] [Q&A] [answered]
Hello,
 I'm using an Arduino based modbus controller to monitor ET and BT in my roaster, connected to my Linux laptop via USB.  This config has been running flawlessly on both windows as well as Linux until the latest update - I've been using it for about 2 years now, upgrading Artisan as the new versions came out.  With v2.6.0, however, the comms take place very intermittently.  My BT and ET would eventually appear, and then disappear with a "modbus communication error" message, and then reappear again after 10-15 seconds for two or three seconds, disappear, and so it repeats.  
I've played with the timeout settings, as well as sample rate, but it doesn't seem to have any effect.  Running 2.4.6 again works like it did.  I work with AppImages, so it's easy to switch.
So, is there anything that was obviously changed in the way modbus comms is handled between 2.4.6 and 2.6.0 that could affect this, and what further info can I post to help?  I'm happy to change things in the arduino code as well if that is what is required, although I haven't touched the code since I got it going originally.  I'm not opening this as an issue, as I'm really not sure it is one.  If the code was just changed to be less forgiving on messy modbus implementation, I'm happy to accept that the problem is my side!  :)
Regards

**Comment by MAKOMO** [comment]
This is a difficult one. The MODBUS implementation was not change a bit. However, the underlying Python lib pymodbus was updated from v2.5.2 to v.2.5.3. The changelog for that release does not ring anything that could be related to your issue.

**Reply by BruciusL** [reply]
Hi Makomo,
 Thanks for getting back to me.  Especially if there isn't suddenly a wave of people having MODBUS communication issues, I'm happy to accept it's something in my code that perhaps has been borderline from the start.  I'll have a look to see if I can find trouble.  Are there any specific debugging switches I can use to have a look at the actual comms taking place?

**Comment by MAKOMO** [comment]
Communication (incl. MODBUS) is logged in under Menu Help >> Serial; errors under Menu Help >> Errors.
Click the plus icon while holding OPTION+COMMAND (macOS) or ALT+CTRL (Windows) to toggle debug logging. The log file is added to an email by clicking the plus icon while holding OPTION/ALT key. The artisan.log file on Linux is located in the directory ~/.local/shared/Artisan-Scope/Artisan. Most helpful would be to run Artisan from source and turn some pymodbus logging on. You have to check the documentation of pymodbus.

**Comment by BruciusL** [comment, marked as answer]
Ok, so after some playing around I can finally report that the problem was in my code. There was a loop previously used for control of the roaster, and removing those and any delays associated with that did the trick. As a bonus I can sample a bit quicker now, so my curves should look a bit better.  
No idea why the problem only started from 2.6.0, but at least it's working now. :)  I can only conclude that the protocol timing was borderline, and something just pushed it over the edge.
